server-mux.go

```diff
@@ -212,3 +280,83 @@ func longestWord(strings []string) int {
 
 	return maxLen
 }
+
+// Close initiates the graceful shutdown
+func (m *MuxServer) Close() error {
+	m.mu.Lock()
+	if !m.closed {
+		m.Server.SetKeepAlivesEnabled(false)
+		m.closed = true
+		m.mu.Unlock()
+		if err := m.listener.Close(); err != nil {
+			return err
+		}
+
+		// force connections to close after timeout
+		wait := make(chan struct{})
+		go func() {
+			m.mu.Lock()
+			for c, st := range m.conns {
+				// Force close any idle and new connections.
+				if st == http.StateIdle || st == http.StateNew {
+					c.Close()
+				}
+			}
+			m.mu.Unlock()
+
+			// Wait for all connections to be gracefully closed
+			m.wg.Wait()
+			close(wait)
+		}()
+
+		select {
+		case <-time.After(m.GracefulTimeout):
+			return nil
+		case <-wait:
+			return nil
+		}
+	} else {
```

Comment:
I guess the logic would be reverted to avoid the 'if block' indentation. 

```
if m.closed {
   return errors.New("Server has been closed")
}
...
... the rest
```